### PR TITLE
refactor: Use react query on NewSubmission form

### PR DESCRIPTION
**Pull request by sgoycoechea-lightit**
Refactored the New Submission form to use the react-query library.

### CODE REVIEW

File: src/screens/NewSubmission.tsx [hunk 1]
```diff
@@ -43,25 +43,28 @@ export default function NewSubmissionScreen({ navigation }: DrawerScreenProps<Dr
...
-      .catch(error => {
-        console.log(error.response);
-        const message = getErrorMessage(error);
-        setApiError(message);
-      }).finally(() => {
-        setIsLoading(false);
+
+  const createSubmission = (data: NewSubmissionFormData) => axiosInstance.post('/submissions', data);
+
+  const { mutate: createSubmissionMutation, isLoading } = useMutation({
```
Comment (ddoumecq) [inline src/screens/NewSubmission.tsx]: you can destructure the mutation error here 
```suggestion
  const { mutate: createSubmissionMutation, isLoading, error } = useMutation({
```

Comment (ddoumecq) [reply src/screens/NewSubmission.tsx]: this way you no longer need `const [apiError, setApiError] = useState<string | null>(null);`

File: src/screens/NewSubmission.tsx [hunk 2]
```diff
@@ -43,25 +43,28 @@ export default function NewSubmissionScreen({ navigation }: DrawerScreenProps<Dr
...
+    mutationFn: createSubmission,
+    onSuccess: () => {
+      setApiError(null);
+      navigation.navigate('HomeStack', {
+        screen: 'Home',
+        params: { newSubmissionAdded: true },
       });
+      reset();
```
Comment (ddoumecq) [inline src/screens/NewSubmission.tsx:55]: if you're navigating you no longer need to reset the form right?

Comment (sgoycoechea-lightit) [reply src/screens/NewSubmission.tsx:55]: Even if I navigate, if I don't reset it, for some reason when I go back the inputs still have what I wrote before

**Review by ddoumecq — APPROVED** [review]

**Pull request merged**